### PR TITLE
stylo: Use the bindgen rewrite to generate the bindings.

### PR DESCRIPTION
**Pull request by emilio**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

- [x] These changes do not require tests because stylo ci is not there yet.

r? @Manishearth

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13269)

### CODE REVIEW

**Comment by emilio** [comment]
cc @upsuper @heycam, this allows us to bindgen `nsINode` and `Element`, so we can remove a lot of out of line calls for things like the flags, the parent, etc...

**Comment by upsuper** [comment]
One `extern "C"` block for all functions becoming one block per function looks unfortunate.

**Comment by emilio** [comment]
@upsuper that's right, but it's a consequence of how the new bindgen code is structured (expects each item to generate its own rust ast item). This simplifies a lot of things, and we could definitely coalesce them, but I think that's not a huge priority.

File: components/style/properties/gecko.mako.rs [hunk 1]
```diff
@@ -1640,7 +1640,7 @@ clip-path
                 fn init_shape(clip_path: &mut StyleClipPath, ty: StyleBasicShapeType) -> &mut StyleBasicShape {
                     unsafe {
                         // We have to be very careful to avoid a copy here!
-                        let ref mut union = clip_path.StyleShapeSource_nsStyleStruct_h_unnamed_26;
+                        let ref mut union = clip_path.__bindgen_anon_1;
```
Comment (Manishearth) [inline components/style/properties/gecko.mako.rs]: yay

**Review by Manishearth — APPROVED** [review]
Looks good to me. Haven't looked too closely at the generated structs, but everything else makes sense.

**Comment by Manishearth** [comment]
(r=me when you need it)

**Comment by emilio** [comment]
The bindgen rewrite landed today.

@bors-servo: r=Manishearth

**Comment by emilio** [comment]
@bors-servo: retry

 * https://github.com/servo/servo/issues/13359

**Comment by emilio** [comment]
Updated the bindgen script to run with `-DMOZ_STYLO` (whoops), and use the latest fixes from https://github.com/servo/rust-bindgen/pull/58.

@bors-servo: r=Manishearth

**Comment by emilio** [comment]
@bors-servo: r=Manishearth

Pushed an OSX-only bindgen fix, as requested by @bholley.

**Pull request merged**